brave/brave-core
# Include FEVM addresses when sender is f1

Resolves https://github.com/brave/brave-browser/issues/30403

When users selects Filecoin there is possibility to choose FEVM address. Translated form will be shown in the chooser and also warning will be shown after FEVM account is selected or typed

![image](https://github.com/brave/brave-core/assets/106654560/bccb3cf5-fae2-446e-a2d1-05fccf3bffaf)

![image](https://github.com/brave/brave-core/assets/106654560/b1a3752a-1ceb-48bb-ac61-cd6bb5d1e51b)



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1) Create Filecoin account, get some funds
2) Select this account on the send screen and select ETH account as recepient. Check that chooser contains FVM address (f410*) for such account.
3) Same thing, but try to type account manually to the recipient field.
4) Check that we have warning which shows actual f410 address for the transaction
5) Complete transaction, check that correct address is used.
6) Check that ETH account received funds on FEVM network.
7) Check same steps for Filecoin testnet\FEVM testnet
